apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/JobsTest.cs

```diff
@@ -67,5 +69,210 @@ public void ListJobs_FullProjection()
 
             Assert.NotNull(jobFound.Resource.Configuration);
         }
+
+        [Fact]
+        public void ListJobs_JobLabels()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var jobToFind = client.CreateQueryJob($"SELECT * FROM {table}", null, new QueryOptions() { Labels = JobLabels});
+
+            // Find the job after listing with full projection.
+            var jobFound = (from job in client.ListJobs(new ListJobsOptions() { Projection = ProjectionEnum.Full })
+                            where job.Reference.JobId == jobToFind.Reference.JobId
+                            select job).Single();
+
+            VerifyJobLabels(jobFound.Resource.Configuration.Labels);
+        }
+
+        [Fact]
+        public void GetJob()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var jobToFind = client.CreateQueryJob($"SELECT * FROM {table}", parameters: null);
+
+            var jobFound = client.GetJob(jobToFind.Reference);
+
+            Assert.Equal(jobToFind.Reference.JobId, jobFound.Reference.JobId);
+        }
+
+        [Fact]
+        public void GetJob_JobLabels()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions() { Labels = JobLabels };
+            var jobToFind = client.CreateQueryJob($"SELECT * FROM {table}", null, options);
+            
+            var jobFound = client.GetJob(jobToFind.Reference);
+
+            VerifyJobLabels(jobFound?.Resource?.Configuration.Labels);
+        }
+
+        [Fact]
```

Comment:
I've generally omitted integration tests if I have a snippet that tests the same thing. (Basically I regard snippets as integration tests in their own right.) I can't easily check right now, but do these tests check anything new? If so, they're very welcome. If they're effectively duplicating snippets, I think we're better off without them.

Reply:
Removed integration tests that were duplicating snippets. Left 3 of them since they are going a little further in testing than the corresponding snippets are. I'm adding comments on this PR for the three tests that remained to signal what they are adding.